icu4c/source/i18n/unicode/coleitr.h

```diff
@@ -242,6 +242,19 @@ class U_I18N_API CollationElementIterator U_FINAL : public UObject {
     */
     void setText(const UnicodeString& str, UErrorCode& status);
 
+#ifndef U_HIDE_INTERNAL_API
+    /**
+    * Sets the source string. Same as the above setText method, but if readOnly
+    * is true then it uses a read-only alias of the input str string.
+    * The str string must outlive this object's lifetime.
+    * @param str the source string.
+    * @param readOnly use a read-only alias of the str string.
+    * @param status the error code status.
+    * @internal
+    */
+    void setText(const UnicodeString &str, UBool readOnly, UErrorCode &status);
```

Comment:
Please rename this function to `internalSetText(...)`.

Actually, I think it would be nicer to have separate functions, `setText()` which continues to make a copy, and `internalSetTextReference(string, status)` which does the fastCopyFrom() keeping the alias. More readable than a boolean parameter.